lib/grover/js/processor.cjs

```diff
@@ -150,7 +150,9 @@ const _processPage = (async (convertAction, urlOrHtml, options) => {
     }
 
     const waitUntil = options.waitUntil; delete options.waitUntil;
-    if (urlOrHtml.match(/^http/i)) {
+    const allowFileUrl = options.allowFileUrl; delete options.allowFileUrl;
+    const urlRegex = allowFileUrl ? /^(http|file)/i : /^http/i;
+    if (urlOrHtml.match(urlRegex)) {
```

Comment:
ok, so this still allows someone to upload say a text file (or some other means of controlling the response body) containing something like:
```
file:///etc/passwd
#<html><meta name='grover-allow_file_url' content='true' /></html>
```
The middleware will subsequently load it as a string, then tell Grover to render it, and your change would allow it to thus exfiltrate the contents of the underlying filesystem by rendering the contents of the file to the PDF

One part of the fix would be to update the `url_source?` method in `OptionsBuilder` to also include the `file` URI scheme. 

It might also be prudent to have some sort of warning in the `OptionsBuilder` that will dump out a stern warning if someone tries to set this new option within `Grover.configuration.options` to at least make it clear that it is exceptionally dangerous.. or better still just exclude this setting from the options to prevent it all together.

Reply:
FYI you can test this on the console with something like:
```
Grover.new("file:///etc/networks\n#<html><meta name='grover-allow_file_url' content='true' /></html>").to_pdf('foo.pdf')
```